lupa/_lupa.pyx

```diff
@@ -571,8 +571,7 @@ cdef class _LuaObject:
             unlock_runtime(self._runtime)
 
     @cython.final
-    cdef inline int push_lua_object(self) except -1:
-        cdef lua_State* L = self._state
+    cdef inline int push_lua_object(self, lua_State* L) except -1:
         lua.lua_rawgeti(L, lua.LUA_REGISTRYINDEX, self._ref)
```

Comment:
That's not true for threads. When testing this function, I found that if we use `self._state` instead of the `L` parameter, some coroutine tests fail.

Reply:
It's easy to reproduce: find calls to `lua_rawgeti` (outside `_LuaIter`) and instead of passing `L`, pass `_LuaObject._state`.

Reply:
All right, good to know!